app/validators/questions/calculated_question_validator.py

```diff
@@ -55,10 +56,27 @@ def validate_calculations_value_source_is_numeric(self):
 
             if answer_id := calculation.get("answer_id"):
                 self._validate_answer_is_numeric(answer_id)
+                self._validate_answers_are_same_numeric_type(
+                    calculation.get("answers_to_calculate"), answer_id
+                )
 
             elif isinstance(value, dict) and value.get("source"):
                 answer_id = value.get("identifier")
                 # Calculated summary value source is validated elsewhere and must be of a number type
 
                 if value.get("source") == "answers":
                     self._validate_answer_is_numeric(answer_id)
+
+    def _validate_answers_are_same_numeric_type(self, answers_to_calculate, answer_id):
+        types = [self.schema.get_answer_type(answer_id)]
+        types.extend(
+            self.schema.get_answer_type(answer) for answer in answers_to_calculate
+        )
+        types_set = set(types)
```

Comment:
this could be simplified with a set comprehension
```
types = {
 self.schema.get_answer_type(answer) for answer in [*answers_to_calculate, answer_id]
}
```